root/etc/cont-init.d/98-ros2

```diff
@@ -0,0 +1,19 @@
+!/usr/bin/with-contenv bash
+
+
+if ! dpkg -l | grep gnupg > /dev/null; then
+    apt-get update && apt-get install -y gnupg
+fi
+
+if [ ! -f "/etc/apt/sources.list.d/ros2.list" ]; then
```

Comment:
repo addition should be moved to a separate init before `95-apt-get` like `94-ros2-src` similar to here: https://github.com/linuxserver/docker-mods/blob/code-server-nodejs/root/etc/cont-init.d/94-nodejs-src